python/ray/data/_internal/datasource/hudi_datasource.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from functools import partial
+from typing import TYPE_CHECKING, Dict, Iterator, List, Optional
+
+from ray.data._internal.util import _check_import
+from ray.data.block import BlockMetadata
+from ray.data.datasource.datasource import Datasource, ReadTask
+from ray.util.annotations import DeveloperAPI
+
+if TYPE_CHECKING:
+    import pyarrow
+
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
```

Comment:
we are planning on dropping usage of the `@DeveloperAPI` label, e.g. https://github.com/ray-project/ray/pull/46825. could you remove it here? thanks!